components/home-page/newsletter-sign-up/NewsletterSignUp.js

```diff
@@ -25,10 +24,33 @@ export default function NewsletterSignUp({
 }) {
   const [isPrivacyPolicyOpen, setIsPrivacyPolicyOpen] = useState(false);
   const [isConfirmationOpen, setIsConfirmationOpen] = useState(false);
+  const [fields, setFields] = useState({});
   const [name, updateName] = useState('');
+  const [errors, setErrors] = useState({});
 
   const onSubmitClick = () => {
-    setIsConfirmationOpen(true);
+    if (isValidSubmission()) {
+      setIsConfirmationOpen(true);
+      setFields({});
+    }
+  };
+
+  const isValidSubmission = () => {
+    let errors = {};
+    let formIsValid = true;
+    const validEmail = new RegExp(
+      '^[a-zA-Z0-9._:$!%-]+@[a-zA-Z0-9.-]+.[a-zA-Z]$',
+    );
+    if (fields['name'] == undefined || fields['name'].length === 0) {
+      formIsValid = false;
+      errors['name'] = 'NAME IS REQUIRED!';
+    }
+    if (!validEmail.test(fields['email'])) {
+      formIsValid = false;
+      errors['email'] = 'EMAIL IS REQUIRED!';
```

Comment:
This is a minor nit: I think `EMAIL IS REQUIRED` should be just if it's empty, `INVALID EMAIL` should be if it doesn't match the regex